tests/yam/agama/patch_agama.pm

```diff
@@ -0,0 +1,35 @@
+## Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: This module use yupdate patch the Agama on Live Medium
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+
+sub run {
+    assert_screen('agama_product_selection', 120);
+
+    select_serial_terminal;
+
+    my ($repo, $branch) = split /#/, get_var('YUPDATE_GIT');
+    if ($repo && $branch) {
+        assert_script_run("yupdate patch $repo $branch", timeout => 60);
+        assert_script_run("playwright test --trace on --project chromium --config /usr/share/e2e-agama-playwright/ test.spec", timeout => 120);
+    } else {
+        record_info("Patch agama SKIP!");
+    }
+}
+
+sub pre_run_hook {
+    $testapi::password = 'linux';
+}
+
+sub post_fail_hook {
+    upload_logs('./test-results/test-The-main-page-has-the-Agama-title-chromium/trace.zip');
```

Comment:
could we create a base class like we did for auto-installation? these methods should not be in test layer.